client/packages/common/src/ui/components/inputs/Filters/FilterRoot.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { FC, useState } from 'react';
+import { CloseIcon, SearchIcon } from '@common/icons';
+import { useUrlQuery } from '@common/hooks';
+import { InlineSpinner } from '../../loading';
+import { Box } from '@mui/material';
+import {
+  IconButton,
+  InputAdornment,
+  DropdownMenu,
+  DropdownMenuItem,
+  Divider,
+} from '@common/components';
+import { useTranslation } from '@common/intl';
+import { TextFilter, TextFilterDefinition } from './TextFilter';
+import { EnumFilter, EnumFilterDefinition } from './EnumFilter';
+
+export interface FilterDefinitionCommon {
+  name: string;
+  urlParameter: string;
+}
+
+type FilterDefinition = TextFilterDefinition | EnumFilterDefinition;
+
+interface FilterDefinitions {
+  filters: FilterDefinition[];
+}
+
+// CONSTANTS
+const RESET_KEYWORD = 'RESET';
+export const FILTER_WIDTH = '220px';
+
+export const FilterRoot: FC<FilterDefinitions> = ({ filters }) => {
+  const t = useTranslation();
+  const { urlQuery, updateQuery } = useUrlQuery();
+  const [activeFilters, setActiveFilters] = useState<FilterDefinition[]>(
+    filters.filter(fil => Object.keys(urlQuery).includes(fil.urlParameter))
+  );
+
+  const filterOptions = getFilterOptions(filters, activeFilters);
+
+  const handleSelect = (selected: string) => {
+    if (selected === RESET_KEYWORD) {
+      const queryPatch = Object.fromEntries(
+        activeFilters.map(({ urlParameter }) => [urlParameter, ''])
+      );
+      updateQuery(queryPatch);
+      setActiveFilters([]);
+      return;
+    }
+    const selectedFilter = filters.find(fil => fil.urlParameter === selected);
+    if (selectedFilter)
+      setActiveFilters(current => [...current, selectedFilter]);
+  };
+
+  const removeFilter = (filterDefinition: FilterDefinition) => {
+    const newActiveFilters = activeFilters.filter(
+      fil => fil.urlParameter !== filterDefinition.urlParameter
+    );
+    updateQuery({ [filterDefinition.urlParameter]: '' });
+    setActiveFilters(newActiveFilters);
+  };
+
+  return (
+    <Box display="flex" gap={2} sx={{ alignItems: 'flex-end', minHeight: 50 }}>
```

Comment:
`minHeight` prevent a UI "jump" when instantiating the first filter, as those components are a bit taller than this root component.